trustmanager/keyfilestore.go

```diff
@@ -301,10 +302,14 @@ func removeKey(s LimitedFileStore, cachedKeys map[string]*cachedKey, name string
 
 // Assumes 2 subdirectories, 1 containing root keys and 1 containing tuf keys
 func getSubdir(alias string) string {
-	if alias == "root" {
+	switch {
+	case alias == data.CanonicalRootRole:
 		return rootKeysSubdir
+	case data.ValidRole(alias):
```

Comment:
I think we should change this to `data.BaseRoles` so that delegations and canonical roles don't have keys in the same place?